kasa/protocol.py

```diff
@@ -105,20 +107,9 @@ class _XorTransport(BaseTransport):
 
     DEFAULT_PORT = 9999
 
-    def __init__(
-        self,
-        host: str,
-        *,
-        port: Optional[int] = None,
-        credentials: Optional[Credentials] = None,
-        timeout: Optional[int] = None,
-    ) -> None:
-        super().__init__(
-            host,
-            port=port or self.DEFAULT_PORT,
-            credentials=credentials,
-            timeout=timeout,
-        )
+    def __init__(self, *, config: DeviceConfig) -> None:
+        super().__init__(config=config)
+        self._port = config.port or self.DEFAULT_PORT
```

Comment:
If one serializes the config after this, the port would be None? I'm not sure what's the nicest fix, but we probably should update the device config / default to the default port to avoid inconsistencies if the default port is selected?

Reply:
I think having the DeviceConfig passed in then be updated internally is a bit fragile and difficult to test so instead I've renamed the `DeviceConfig` `port` parameter to be `port_override` and implemented a `default_port` property on the `BaseTransport` abstract class.  This makes it a lot simpler to see what is happening with the port in the constructors and it all happens in `BaseTranport.__init__()`: 
  `self._port = config.port_override or self.default_port`

I hope that makes sense.  The `port` implementation was complicated compared to the other fields and I think this makes it much simpler.